apps/server/src/app/opensearch/create-indexes.ts

```diff
@@ -0,0 +1,103 @@
+import { Client } from "@opensearch-project/opensearch";
+import { pino } from "pino";
+
+export async function createIndexes(client: Client, _logger: pino.Logger) {
+  const logger = _logger.child({ step: "createIndexes" });
+  logger.info("Creating indexes");
+  await createRequestsIndex(client, logger);
+}
+
+export async function createRequestsIndex(
+  client: Client,
+  _logger: pino.Logger
+) {
+  const logger = _logger.child({ index: "createRequestsIndex" });
+  const index = "requests";
+
+  const indexExists = await client.indices.exists({
+    index,
+  });
+
+  if (!indexExists.body) {
+    logger.info("Creating index");
+
+    await client.indices.create({
+      index,
+    });
+    return;
+  } else {
+    logger.info("Index already exists, skipping creation");
+  }
+
+  logger.info("Index created");
+  logger.info("Setting/Updating mappings");
+
+  await client.indices.putMapping({
+    index,
+    body: {
+      properties: {
+        ownership: {
+          properties: {
+            organizationId: {
+              type: "keyword",
+            },
+            projectId: {
+              type: "keyword",
+            },
+          },
+        },
+        reportId: {
+          type: "keyword",
+        },
+        calculated: {
```

Comment:
This probably is going to be different per provider

Reply:
This isn't a strict schema, just strictly defines our mapped properties. You can dump anything in the document.